src/System.Threading.Tasks/tests/Task/TaskRtTests.cs

```diff
@@ -0,0 +1,1135 @@
+            Task t4 = new Task<int>(() =>
+            {
+                i4 = new Task(() => { }, TaskCreationOptions.AttachedToParent);
+                return 42;
+            }, TaskCreationOptions.DenyChildAttach);
+            t4.Start();
+
+            // continuations, Task and Future
+            Task i5 = null;
+            Task t5 = t3.ContinueWith(_ =>
+            {
+                i5 = new Task(() => { }, TaskCreationOptions.AttachedToParent);
+            }, TaskContinuationOptions.DenyChildAttach);
+
+            Task i6 = null;
+            Task t6 = t4.ContinueWith<int>(_ =>
+            {
+                i6 = new Task(() => { }, TaskCreationOptions.AttachedToParent);
+                return 42;
+            }, TaskContinuationOptions.DenyChildAttach);
+
+            // If DenyChildAttach doesn't work in any of the cases, then the associated "parent"
+            // task will hang waiting for its child.
+            Debug.WriteLine("RunDenyChildAttachTests: Waiting on 'parents' ... if we hang, something went wrong.");
+            Task.WaitAll(t1, t2, t3, t4, t5, t6);
+
+            // And clean up.
+            i1.Start(); i1.Wait();
+            i2.Start(); i2.Wait();
+            i3.Start(); i3.Wait();
+            i4.Start(); i4.Wait();
+            i5.Start(); i5.Wait();
+            i6.Start(); i6.Wait();
+        }
+
+        [Fact]
+        public static void RunBasicFutureTest_Negative()
+        {
+            Task<int> future = new Task<int>(() => 1);
+            Assert.ThrowsAsync<ArgumentNullException>(
+               () => future.ContinueWith((Action<Task<int>, Object>)null, null, CancellationToken.None));
+            Assert.ThrowsAsync<ArgumentNullException>(
+              () => future.ContinueWith((Action<Task<int>, Object>)null, null, TaskContinuationOptions.None));
+            Assert.ThrowsAsync<ArgumentNullException>(
+              () => future.ContinueWith((Action<Task<int>, Object>)null, null, CancellationToken.None, TaskContinuationOptions.None, TaskScheduler.Default));
+            Assert.ThrowsAsync<ArgumentNullException>(
+              () => future.ContinueWith((t, s) => { }, null, CancellationToken.None, TaskContinuationOptions.None, null));
+
+            Assert.ThrowsAsync<ArgumentNullException>(
+               () => future.ContinueWith<int>((Func<Task<int>, Object, int>)null, null, CancellationToken.None));
+            Assert.ThrowsAsync<ArgumentNullException>(
+              () => future.ContinueWith<int>((Func<Task<int>, Object, int>)null, null, TaskContinuationOptions.None));
+            Assert.ThrowsAsync<ArgumentNullException>(
+              () => future.ContinueWith<int>((Func<Task<int>, Object, int>)null, null, CancellationToken.None, TaskContinuationOptions.None, TaskScheduler.Default));
+            Assert.ThrowsAsync<ArgumentNullException>(
+              () => future.ContinueWith<int>((t, s) => 2, null, CancellationToken.None, TaskContinuationOptions.None, null));
```

Comment:
Ran across these in master. None of these are awaited. Looks like a bug both now and at the time.
https://github.com/dotnet/corefx/blob/3efd19931a28e0d978fb273d20d857f597fca93c/src/System.Threading.Tasks/tests/project.json#L15
https://github.com/xunit/xunit/blob/2.1-beta3/src/xunit.assert/Asserts/ExceptionAsserts.cs#L50

Reply:
Ugh. Yeah, that's busted.  I'll fix it.  Thanks for pointing it out.